modules/swift/examples/code_snippets/SampleCodeTest.swift

```diff
@@ -1209,2824 +1208,1603 @@ class SampleCodeTest {
-        */
-
-
-        // tag::tojson-dictionary[]
-
-        var aJSONstring = """
-            {\"id\":\"1002\",\"type\":\"hotel\",\"name\":\"Hotel Ned\",\"city\":\"Balmain\",
-            \"country\":\"Australia\",\"description\":\"Undefined description for Hotel Ned\"}
-            """
-
-        let myDict:MutableDictionaryObject =
-            try! MutableDictionaryObject(json: aJSONstring) // <.>
-        print(myDict)
-
-        let name = myDict.string(forKey: "name")
-        print("Details for: ", name!)
-
-        for key in myDict {
-
-            print(key, myDict.value(forKey: key) as! String)
-
-        }
-
-
-        // end::tojson-dictionary[]
-
-        /*
-        // tag::tojson-dictionary-output[]
-
-         Details for:  Hotel Ned
-         description Undefined description for Hotel Ned
-         id 1002
-         name Hotel Ned
-         country Australia
-         type hotel
-         city Balmain
-
-         // end::tojson-dictionary-output[]
-        */
-    public func JsonApiBlob() throws {
-        // tag::tojson-blob[]
-      // Get a document
-        let thisDoc = db.document(withID: "1000")?.toMutable() // <.>
-
-        // Get the image and add as a blob to the document
-        let contentType = "image/jpg";
-        let ourImage = UIImage(named: "couchbaseimage.png")!
-        let imageData = ourImage.jpegData(compressionQuality: 1)!
-        thisDoc?.setBlob(
-            Blob(contentType: contentType, data: imageData), forKey: "avatar") // <.>
-
-       let theBlobAsJSONstringFails =
-              thisDoc?.blob(forKey: "avatar")!.toJSON();
-
-        // Save blob as part of doc or alternatively as a blob
-
-        try! db.saveDocument(thisDoc!);
-        try! db.saveBlob(
-                blob: Blob(contentType: contentType, data: imageData)); // <.>
-
-        // Retrieve saved blob as a JSON, reconstitue and check still blob
-        let sameDoc = db.document(withID: "1000")
-        let sameBlob = sameDoc?.blob(forKey: "avatar")
-        let theBlobAsJSONstring = sameBlob!.toJSON(); // <.>
-
-        for (key, value) in sameDoc!.toDictionary() {
-             print( "Data -- {0) = {1}", key, value);
-        }
-
-        if(Blob.isBlob(properties: sameBlob!.properties)) // <.>
-        {
-            print(theBlobAsJSONstring);
-        }
-
-        // end::tojson-blob[]
-
-
-    }
-
-//    } // end func testjson
-
-//        } // end query loop
-
-    } // end jsonapi func
-
-
-}
-
-    extension String {
-
-        func toJSONObj() -> Any {
-
-            let d1 = self.data(using: .utf8)
-
-            return try! JSONSerialization.jsonObject(
-                with: d1!, options:[])
+        
+        for row in results {
+            print(row["thisId"].string!)
+            
+            let docsId = row["thisId"].string!
+            
+            // Now you can get the document using the ID
+            let doc = database.document(withID: docsId)!.toDictionary()
```

Comment:
don't convert to dictionary. I don't remember if Objective-C code is doing this or not.